carbon-design-system/carbon-for-ibm-dotcom
# Fix/mobile masthead csp

### Related Ticket(s)

Fixes #9675 

### Description

This PR continues the work of moving inline styles out of the masthead to ensure CSP compliance within the cloud console. There are some changes to how the CSP compliance mixin is implemented, but the approach of creating a nonced style element within the shadow root is largely the same.

In addition to components creating an internal stylesheet, they also share a global stylesheet for elements that need to utilize it. The internal stylesheet should always be preferred, but some elements, like the `dds-masthead-container` don't attach a shadowroot, and thus cannot have an "internal" stylesheet.


### Changelog

**New**

- Implemented CSP compliance mixin in additional masthead components

**Changed**

- Updates to CSP compliance mixin

### Testing Instructions

The simplest way to test this is to open the chrome console tab of the devtools and to add a "live expression" for `$$('#main-content [style]')`. This will show any elements within the main content wrapper with an inline style attribute. Click through the masthead in various screen widths and ensure no elements populate that live expression.
<img width="253" alt="Screenshot 2022-11-21 at 9 29 42 PM" src="https://user-images.githubusercontent.com/25532785/203205404-a9166e94-a4a6-4bbb-831f-665094d5b567.png">
